Packet++/header/CotpLayer.h

```diff
@@ -0,0 +1,145 @@
+#ifndef PCAPPLUSPLUS_COTPLAYER_H
+#define PCAPPLUSPLUS_COTPLAYER_H
+
+#include "EthLayer.h"
+#include "Layer.h"
+
+namespace pcpp
+{
+
+/**
+ * @struct cotphdr
+ * Represents a COTP protocol header
+ */
+#pragma pack(push, 1)
+	typedef struct
+	{
+		/** length */
+		uint8_t length;
+		/** PDU type identifier */
+		uint8_t pdu_type;
+		/** TPDU number sequence*/
+		uint8_t tpdu_number;
+	} cotphdr;
+#pragma pack(pop)
+
+	/**
+	 * @class CotpLayer
+	 * Represents a COTP (Connection Oriented Transport Protocol)
+	 */
+	class CotpLayer : public Layer
+	{
+	  public:
+		/**
+		 * A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data (will be casted to @ref cotphdr)
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		CotpLayer(uint8_t *data, size_t dataLen, Layer *prevLayer, Packet *packet)
+			: Layer(data, dataLen, prevLayer, packet)
+		{
+			m_Protocol = COTP;
+		}
+
+		/**
+		 * A constructor that allocates a new COTP header
+		 * @param[in] length Packet length
+		 * @param[in] pdu_type Protocol PDU type number
+		 * @param[in] tpdu_number Protocol TPDU number
+		 */
+		CotpLayer(uint8_t length, uint8_t pdu_type, uint8_t tpdu_number);
```

Comment:
1. Params should be camelCase - so `pduType` and `tpduNumber`
2. Do we currently support length != 2 and pdu type != `0x0f`? If not, maybe these parameters shouldn't be part of the constructor? In that case the only parameter should be `tpduNumber`